### DISCUSSION TITLE
Error : uses undefined struct 'asn1_object_st'

### DISCUSSION THREAD

**Discussion by DOpenssl** [openssl/openssl#23694] [Q&A] [answered]
I am compiling OpenSSL1.1.1 with Visual Studio 2017 on windows 10.
I am using below code in my .cpp file
  #include <openssl/ossl_typ.h>
   func(){
       ASN1_OBJECT obj;
   }

I am getting error:

error C2079: 'obj' uses undefined struct 'asn1_object_st'

I have included "openssl/ossl_typ.h" in my file.

I am giving openssl "-IC:\openssl-1.1.1w\include" option to "cl" command

Please let me know how to fix above compilation error..

**Comment by t8m** [comment, marked as answer]
You cannot declare the ASN1_OBJECT as local variable. You have to allocate it on heap via functions from OpenSSL that create it and pass just pointers to it.
